src/charm.py

```diff
@@ -240,13 +240,44 @@ def configure_app_non_leader(self, event: ops.framework.EventBase) -> None:
         if isinstance(event, MicroClusterNodeAddedEvent):
             self.cluster_nodes.join_node_to_cluster(event)
 
+    def _get_space_subnet(self, space: str):
+        """Get the first available subnet in the network space."""
+        space_nets = self.model.get_binding(binding_key=space).network.interfaces
+        if space_nets:
+            return space_nets[0].subnet
+
+    def _get_bootstrap_params(self) -> dict:
+        """Fetch bootstrap parameters."""
+        micro_ip = cluster_net = public_net = ""
+
+        if "quincy" in self.model.config.get("snap-channel"):
+            # all quincy snap revisions do not support network configuration
```

Comment:
Wording nit, suggest: `none of the quincy snap revisions support network configuration`
Also might want to add a logging statement that we're ignoring quincy